api/src/pcapi/repository/__init__.py

```diff
@@ -6,8 +6,9 @@
 @contextmanager
 def transaction():  # type: ignore [no-untyped-def]
     try:
-        yield
-        db.session.commit()
+        with db.session.no_autoflush:
+            yield
+            db.session.commit()
```

Comment:
```suggestion
        with db.session.no_autoflush:
            yield
        db.session.commit()
```
Il me semble plus cohérent d'avoir commit et rollback au même niveau